### PR TITLE
Modernize a few of the files

### PR DESCRIPTION
**Pull request by SivanMehta**
> convert it to a `class`, arrow functions and using `this` over `constructor`, and think about if you want to do the async await conversion or if getting your hands dirty enough

- [x] Convert `prototype`s to use a `class`
- [x] Use `this` instead of `thingy = this` (will also call for arrow functions)

"the async await conversion" will come in a later PR to focus the scope of each set of changes. Right now I'm mostly focusing on syntax.

### CODE REVIEW

File: lib/construct/index.js [hunk 1]
```diff
@@ -56,834 +52,829 @@ const envs = {
...
-
-//
-// Extend Constructor with event emitting capacities.
-//
-util.inherits(Constructor, EventEmitter);
-Constructor.prototype.emits = require('emits');
+class Constructor extends EventEmitter {
+  constructor(app, options) {
+    super();
+    EventEmitter.call(this, options);
```
Comment (jcrugzz) [inline lib/construct/index.js]: is this needed given the `super()` call? It should be equivalent. you could pass `options` though

File: lib/construct/index.js [hunk 2]
```diff
@@ -56,834 +52,829 @@ const envs = {
...
-  app.contextLog.info('compile build spec for %s', data.name, {
-    name: data.name
-  });
-
-  constructor.specs(data, function specifications(error, spec) {
-    if (error) {
-      return void progress.fail(error);
-    }
+  emits() {
+    require('emits')(...arguments);
```
Comment (jcrugzz) [inline lib/construct/index.js]: do we actually use this anywhere explicitly?

**Review by jcrugzz — CHANGES_REQUESTED** [review]

File: lib/construct/index.js (hunk 2 above)
Comment (SivanMehta) [reply lib/construct/index.js]: I'm not sure about in this repo. This function serves to replace the original prototype declaration [here](https://github.com/godaddy/carpenterd/blob/master/lib/construct/index.js#L98
)

```js
Constructor.prototype.emits = require('emits');
```

File: lib/app.js [hunk 3]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (3rd-Eden) [inline lib/app.js]: The `use strict` should be kept for Node.js

**Review by 3rd-Eden — CHANGES_REQUESTED** [review]

File: lib/app.js (hunk 3 above)
Comment (3rd-Eden) [reply lib/app.js]: related: https://twitter.com/matteocollina/status/1100468412701126656

File: lib/construct/index.js (hunk 2 above)
Comment (jcrugzz) [reply lib/construct/index.js]: @SivanMehta understood, id just verify that its used otherwise remove altogether

**Review by jcrugzz — APPROVED** [review]

File: lib/construct/index.js (hunk 2 above)
Comment (3rd-Eden) [reply lib/construct/index.js]: Yeah, let's not call `require` every time this function is called as `require` is a **sync** operation which will freeze the event loop. I would suggest doing a `const emits = require('emits')` and doing a `this.emits = emits` in the `constructor` to restore the original functionality .

**Review by 3rd-Eden — APPROVED** [review]

**Pull request merged**